### PR TITLE
add dp3000 macropad rev2

### PR DESCRIPTION
**Pull request by depermana12**
## Description

* update oled logo
* add rev2 support `rgblight` underglow, and restructure the folder
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dp3000/dp3000.c [hunk 1]
```diff
@@ -14,21 +14,94 @@
...
+    oled_timer = timer_read32();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        oled_timer_reset();
+    }
+
+    return true;
+}
```
Comment (drashna) [inline keyboards/dp3000/dp3000.c:107]: `last_input_activity_time()` or `last_input_activity_elapsed()` are better options here, for checking the timer. 

The keypresses, encoder and pointing device actually trigger this timer, so may be more accurate, when it comes to activity.

File: keyboards/dp3000/rev1/info.json
Comment (drashna) [inline keyboards/dp3000/rev1/info.json:1]: This has a mix of spaces and tabs.  `qmk format-json` will help with this.

File: keyboards/dp3000/rev2/info.json
Comment (drashna) [inline keyboards/dp3000/rev2/info.json:1]: same.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/dp3000/dp3000.c (hunk 1 above)
Comment (depermana12) [reply keyboards/dp3000/dp3000.c]: Thank you for the suggestion. `Using last_input_activity_elapsed()` sounds like more accurate and reliable to measure user activity

**Review by drashna — APPROVED** [review]

File: keyboards/dp3000/rev2/config.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
```
Comment (fauxpark) [inline keyboards/dp3000/rev2/config.h:1]: This file can be deleted.

File: keyboards/dp3000/dp3000.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
```
Comment (fauxpark) [inline keyboards/dp3000/dp3000.h:18]:
```suggestion
#pragma once

#include "quantum.h"
```

File: keyboards/dp3000/dp3000.c [hunk 4]
```diff
@@ -14,21 +14,83 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 #include "quantum.h"
```
Comment (fauxpark) [inline keyboards/dp3000/dp3000.c:16]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by depermana12** [comment]
sorry to summon you @fauxpark, I don't mean to be impatient, but is there anything else need to changes to get approval

**Comment by depermana12** [comment]
Hello, is there are any additional changes required to get this PR accepted? I would appreciate any feedback. Please let me know if there's any issue. Looking forward to your response. Thank you!

**Comment by depermana12** [comment]
Is there anything else I need to do to get PR accepted?

**Comment by depermana12** [comment]
bumping, it's been a while. can i get a code review

**Review by fauxpark — CHANGES_REQUESTED** [review]
Keyboard updates need to go through `develop`.

**Comment by depermana12** [comment]
> Member

I've followed your suggestion, changed from master to develop

**Comment by depermana12** [comment]
hi @fauxpark Please kindly help check my PR again, is there anything else need to change to get this PR merge? because these macropad have ready to ship and need to support via auto-detect.

File: keyboards/dp3000/rev1/info.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            "raindrops": true,
+            "solid_color": true,
+            "solid_reactive": true,
+            "solid_reactive_multinexus": true,
+            "solid_reactive_simple": true,
+            "solid_splash": true,
+            "splash": true,
+            "typing_heatmap": true
+        },
+        "driver": "WS2812",
```
Comment (fauxpark) [inline keyboards/dp3000/rev1/info.json:40]:
```suggestion
        "driver": "ws2812",
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request closed without merging**